bin/smarthome.py

```diff
@@ -184,9 +185,25 @@ def __init__(self, smarthome_conf_basename=os.path.join(_etc_dir,'smarthome')):
         self.version = VERSION
         self.connections = []
 
+        if self._extern_conf_dir is not None:
+            self._etc_dir = os.path.join(self._extern_conf_dir, 'etc')
+            self._items_dir = os.path.join(self._extern_conf_dir, 'items'+os.path.sep)
+            self._logic_dir = os.path.join(self._extern_conf_dir, 'logics'+os.path.sep)
+            self._smarthome_conf_basename = os.path.join(self._etc_dir,'smarthome')
+            self._logic_conf_basename = os.path.join(self._etc_dir, 'logic')
+            self._module_conf_basename = os.path.join(self._etc_dir,'module')
+            self._plugin_conf_basename = os.path.join(self._etc_dir,'plugin')
+            self._log_config = os.path.join(self._etc_dir,'logging'+YAML_FILE)
```

Comment:
I don't like this special handling. This could be avoided if we default `self._extern_conf_dir` to something like `_base_dir` and always do this block.

Reply:
OK, i guess that in this case there's no reason to do it twice (just before `__init__` and inside `__init__`)